invenio_users_resources/services/schemas.py

```diff
@@ -100,7 +101,7 @@ class UserSchema(BaseRecordSchema, FieldPermissionsMixin):
     visibility = fields.Str(dump_only=True)
     is_current_user = fields.Method("is_self", dump_only=True)
 
-    email = fields.String()
+    email = fields.Email()
```

Comment:
While this is correct, it introduces stricter validation. Existing users with invalid email formats might fail schema validations. To prevent disruptions, a migration strategy should be planned to handle or correct existing invalid email entries.

Reply:
That's an excellent point. I'm unsure how you'd get wrongly formatted emails on the db, though.

There are plenty of other places where wrong emails might break stuff. Off the top of my head, some of the domain-related stuff, probably https://github.com/inveniosoftware/invenio-accounts/blob/master/invenio_accounts/utils.py#L264
I don't know what the best path forward is. If we decide to do a "migration", what do we do with wrongly formatted emails? (email is a required column on the DB). 
Let's get someone else's opinion on this.